extensions/amp-script/0.1/user-activation-tracker.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+// TODO: move to utils/?
+
+export const ACTIVATION_TIMEOUT = 5000;  // 5 seconds.
+
+const ACTIVATION_EVENTS = ['click', 'input', 'dblclick', 'keypress', 'submit'];
+
+
+/**
+ * See https://github.com/dtapuska/useractivation for inspiration.
+ */
+export class UserActivationTracker {
```

Comment:
Do you plan to expose this in worker scope too, or is this just an AMP implementation detail?

Reply:
The very first step: only inside AMP. However, I was going to talk to you and Dave about it. And if we are feeling confident - I was going to polyfill UserActivation on the Web Worker's navigator object. WDYT?

Reply:
Scoping is different but maybe just replacing it is fine. The override should only apply in the "safe" variant for workerDOM (maybe we should rename "safe" to "amp").